### PR TITLE
Phase 1

### PR DESCRIPTION
**Pull request by ben-gibson**

### CODE REVIEW

File: src/Domain/Poll/PollRepository.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+    public function __construct(EventStreamStore $eventStreamStore, EventBus $eventBus)
+    {
+        $this->eventStore = $eventStreamStore;
+        $this->eventBus   = $eventBus;
+    }
+
+
+    public function save(Poll $poll) : void
```
Comment (ben-gibson) [inline src/Domain/Poll/PollRepository.php:22]: Thinking about this a little more, it's going to miss events raised by non aggregate roots.

File: src/UI/CLI/Bus/ConsumeEvents.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    public function __construct()
+    {
+        parent::__construct('pollhq:bus:consume:events');
+    }
+
+
+    protected function execute(InputInterface $input, OutputInterface $output) : void
+    {
+        $output->writeln('Consuming events.');
```
Comment (ben-gibson) [inline src/UI/CLI/Bus/ConsumeEvents.php:22]: Events are currently based through the event bus in the same process as the command handling however they could be consume in a separate process.

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     * ENTRY CONFIG
+     *
+     * Add 1 entry for each "page" of your app
+     * (including one that's included on every page - e.g. "app")
+     *
+     * Each entry will result in one JavaScript file (e.g. app.js)
+     * and one CSS file (e.g. app.scss) if you JavaScript imports CSS.
+     */
+    .addEntry('app-dark', './assets/js/app-dark.js')
+    .addEntry('app-light', './assets/js/app-light.js')
```
Comment (krixon) [inline webpack.config.js]: This is going to bundle two versions of the JS too - might be better to bundle everything together and determine which theme is active by adding a class to the body element dynamically. The theme customisations can then be scoped under that class.

File: src/Adapter/Persistence/Doctrine/DBALEventStreamStore.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    public function appendEvents(EventStream $eventStream) : void
+    {
+        foreach ($eventStream->events() as $event) {
+            $this->appendEvent($eventStream->id(), $event);
+        }
+    }
+
+
+    private function appendEvent(EventStreamId $id, BaseEvent $event) : void
```
Comment (krixon) [inline src/Adapter/Persistence/Doctrine/DBALEventStreamStore.php]: Could this use the `Event` interface instead?

File: src/Adapter/Persistence/Doctrine/DBALEventStreamStore.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        if (!($event instanceof Event)) {
+            throw new InvalidArgument(sprintf("Expected an instance of '%s'", Event::class));
```
Comment (krixon) [inline src/Adapter/Persistence/Doctrine/DBALEventStreamStore.php]: This seems redundant - `BaseEvent` is always an instance of `Event`.

File: config/packages/doctrine.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+parameters:
+    # Adds a fallback DATABASE_URL if the env var is not set.
+    # This allows you to run cache:warmup even if your
+    # environment variables are not available yet.
+    # You should not need to change this value.
+
+doctrine:
+    dbal:
+        connections:
+            default:
```
Comment (ben-gibson) [inline config/packages/doctrine.yaml]: This will be separated into two databases. One for the events and the other for read models.

**Pull request merged**